File: src/list/get_desired_sublist_sum.ts
// ref: https://github.com/soygul/QuanticDev/blob/master/algorithms/dynamic-programming/sliding-window/desired-subarray-sum.js

export default function getDesiredSublistSum(input: number[], desiredSum: number) {
  const output: Array<number[]> = [];
  let sum = 0;
  let sumStartIndex = 0;
  for (let index = 0; index < input.length; index++) {
    const number = input[index];
    sum += number;
    while (sum > desiredSum) {
      sum -= input[sumStartIndex];
      sumStartIndex++;
    }
    if (sum === desiredSum) output.push(input.slice(sumStartIndex, index + 1));
  }
  return output;
}

File: src/list/get_desired_sublist_sum.py
def get_desired_sublist_sum(arr: list[int], desired_sum: int):
  output: list[list[int]] = []
  count = 0
  sum_start_index = 0
  for index, element in enumerate(arr):
    count += element
    while count > desired_sum:
      count -= arr[sum_start_index]
      sum_start_index += 1
    if count == desired_sum:
      output.append(arr[sum_start_index:index + 1])
  return output
